dm/config/security_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"bytes"
+	"os"
+
+	. "github.com/pingcap/check"
+)
+
+const (
+	testdataPath = "./testdata"
+
+	caFile        = "./testdata/ca.pem"
+	caFileContent = `
+-----BEGIN CERTIFICATE-----
+test no content
+-----END CERTIFICATE-----
+`
+	certFile        = "./testdata/cert.pem"
+	certFileContent = `
+-----BEGIN CERTIFICATE-----
+test no content
+-----END CERTIFICATE-----
+`
+	keyFile        = "./testdata/key.pem"
+	keyFileContent = `
+-----BEGIN RSA PRIVATE KEY-----
+test no content
+-----END RSA PRIVATE KEY-----
+`
+)
+
+func createTestFixture(c *C) {
+	c.Assert(os.Mkdir(testdataPath, 0o744), IsNil)
+
+	f, err := os.Create(caFile)
+	c.Assert(err, IsNil)
+	_, err = f.WriteString(caFileContent)
+	c.Assert(err, IsNil)
+	f.Close()
+
+	f, err = os.Create(certFile)
+	c.Assert(err, IsNil)
+	_, err = f.WriteString(certFileContent)
+	c.Assert(err, IsNil)
+	f.Close()
+
+	f, err = os.Create(keyFile)
+	c.Assert(err, IsNil)
+	_, err = f.WriteString(keyFileContent)
+	c.Assert(err, IsNil)
+	f.Close()
+}
+
+func clearTestFixture(c *C) {
+	c.Assert(os.RemoveAll(testdataPath), IsNil)
+}
+
+func (t *testConfig) TestLoadAndClearContent(c *C) {
+	createTestFixture(c)
+	defer clearTestFixture(c)
+
+	s := &Security{
+		SSLCA:   "testdata/ca.pem",
```

Comment:
could it use `caFile`?

if so, also remember to fix rest of them

Reply:
Probably not, because I want to test whether LoadTLSContent is working as expected

Reply:
Probably not, because I want to test whether LoadTLSContent is working as expected

Reply:
fixed in https://github.com/pingcap/dm/pull/1910/commits/fff653c01bb174e8824a5323e750dab9263597a3